pkg/filters/grpc_sidecar_filter.go

```diff
@@ -55,59 +60,63 @@ func (f *grpcFilterType) Configure(data map[string]interface{}) error {
 		return errors.Errorf("empty binary-url")
 	}
 
-	// name is the binary name. when we downloaded the binary, we use this name to
+	// name is the binary name. when we download the binary, we use this name to
 	// launch the process
 	name, ok := data["name"]
 	if !ok {
 		return errors.Errorf("empty binary name")
 	}
 
-	urlString := url.(string)
-	nameString := name.(string)
-
-	fileName, err := utils.GetExecutable(urlString, BinaryDir, nameString)
-	if err != nil {
-		return errors.Trace(err)
-	}
-
-	// start the process
-	client := hplugin.NewClient(&hplugin.ClientConfig{
-		HandshakeConfig: grpc.HandshakeConfig,
-		Plugins:         grpc.PluginMap,
-		Cmd:             exec.Command(fileName),
-		AllowedProtocols: []hplugin.Protocol{
-			hplugin.ProtocolGRPC,
-		},
-	})
-
-	rpcClient, err := client.Client()
-	if err != nil {
-		return errors.Trace(err)
-	}
-
-	f.client = client
-
-	raw, err := rpcClient.Dispense(grpc.PluginName)
-	if err != nil {
-		return errors.Trace(err)
-	}
-
-	delegate := raw.(core.IFilter)
-	f.delegate = delegate
+	f.configData = data
+	f.binaryURL = url.(string)
+	f.name = name.(string)
 
-	err = f.ConfigureMatchers(data)
+	err := f.ConfigureMatchers(data)
 	if err != nil {
 		return errors.Trace(err)
 	}
-
-	if err := f.delegate.Configure(data); err != nil {
-		return errors.Trace(err)
-	}
-
 	return nil
 }
 
 func (f *grpcFilterType) Filter(msg *core.Msg) (bool, error) {
+
+	if f.client == nil {
```

Comment:
this is not thread safe. wrap it with sync.once

Reply:
there will be only one emitter in the system that will call filter functions.

Reply:
no. mongo-batch called emitter concurrently.

Reply:
PTAL @Ryan-Git

Reply:
@Ryan-Git PTAL